src/reader/ArticleReader.js

```diff
@@ -108,10 +107,22 @@ export default function ArticleReader({ api, teacherArticleID }) {
   };
 
   const handleScroll = () => {
-    onScroll(api, articleID, UMR_SOURCE);
+    let percentage = Math.floor(updateScrollPosition() * 100);
```

Comment:
For me, the code would be easier to read if we wrote it like: 

```js 
let ratio = getScrollRatio();
setScrollRatio(ratio);
let percentage = ratio * 100; 

```

the main difference is that in this approach the `get` function is a pure computation function as opposed to the `update` function which does two things: compute and set and then return the computation. 

Do you see what I mean?

Reply:
Agreed